apps/dex/src/pages/swap.tsx

```diff
@@ -370,25 +372,36 @@ const SwapPage = () => {
     toDenom,
   ]);
 
-  const swapButtonMsg = (() => {
+  const validationError = useMemo(() => {
     if (signerStatus === "resolved" && signer === undefined) {
-      return "Please Connect Sif Wallet";
+      return new Error("Please Connect Sif Wallet");
     }
 
-    if (!isReady) {
-      return "Loading";
+    if (fromAmountDecimal?.toFloatApproximation() === 0) {
+      return new Error();
     }
 
     if (
       fromAmountDecimal?.isGreaterThan(
         fromToken?.balance ?? Decimal.zero(fromAmountDecimal.fractionalDigits),
       )
     ) {
-      return "Insufficient balance";
+      return new Error("Insufficient balance");
```

Comment:
same here, we could have an constant error file and have finite states for all of them